### PR TITLE
docs: describe intended ways to use Renovate

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Describe intended way to use Renovate in our `design-decisions.md` docs
- ~Add section to the FAQ for end-users that describes the intended ways to use Renovate bot~ (Decided to drop this idea!)

## Context:

Me and @viceice discussed the Renovate API not being stable (https://github.com/renovatebot/renovate/pull/11147).
I discovered we basically don't mention our API being unstable anywhere, we also don't really say to the end-user "Here's the supported ways to use Renovate".

@viceice made the point that it's undocumented for a reason: so that people don't go use the internals of Renovate for their own thing. Also documenting the API is a lot of work, and those API docs would need to be updated regularly.
Basically their point boils down to: "We don't document it anywhere, so people can't use it in bad ways."

I think we should at least mention somewhere suitable that our API is meant for our use only, and that people should only use Renovate as an app. We should also say which ways of using Renovate as an app are supported.

Maybe I'm honking up the wrong tree here, and the Renovate maintainers are happy to leave things as they are. 😉 
I'll take the risk of a rejected PR, as my point will be clearer to see with a PR than with a new issue that re-hashes the discussion. 😄 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/design-decisions.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 This file documents the design choices as well as configuration options.
 
+## Intended usage by end-users
+
+Renovate is intended to be used as an app.
+It should not be used as a library, because it lacks a stable API.
+
+We do not want to create a stable API, or document our existing API.
```
Comment (viceice) [inline docs/development/design-decisions.md:10]: We want a stable api some time, but currently it's changing too often to be stable soon.

**Comment by HonkingGoose** [comment]
I have a feeling this PR will be superseded by #11202 that @rarkins is working on... 😄 

I'll let @rarkins decide if they want this PR or not. Or maybe they can incorporate the relevant changes in their own PR. 😉

**Comment by rarkins** [comment]
I think this is correctly placed in the "development" side of the docs and complementary

File: docs/development/design-decisions.md (hunk 1 above)
Comment (HonkingGoose) [reply docs/development/design-decisions.md:10]: Do we want to make any promises/future work by suggesting we're going to stabilize our API? 

@rarkins What do you think?

File: docs/usage/faq.md [hunk 2]
```diff
@@ -39,6 +39,17 @@ The Renovate team will only create bugfixes for an older version if the hosted a
...
+
+Renovate is intended to be used as an app.
+We do not support using Renovate as a library.
+
+Supported ways to use Renovate:
+
+- via the CLI program
+- via the official hosted app
+- via selfhosting the Renovate app
+
```
Comment (HonkingGoose) [inline docs/usage/faq.md:52]: @rarkins Do you want to keep this section in the FAQ? I think it's a bit superfluous now that we have a full getting started guide with recommended ways to use Renovate in it.

Or we could just link to the new getting started guide?

Comment (rarkins) [reply docs/usage/faq.md:52]: Yes, we can remove it. Most of the FAQ hasn't been updated in a very long time!

Comment (HonkingGoose) [reply docs/usage/faq.md:52]:
```suggestion
```

File: docs/development/design-decisions.md [hunk 3]
```diff
@@ -2,6 +2,17 @@
 
 This file documents the design choices as well as configuration options.
 
+## Intended usage by end-users
```
Comment (HonkingGoose) [inline docs/development/design-decisions.md:5]:
```suggestion
## Intended use by end-users
```

File: docs/development/design-decisions.md (hunk 1 above)
Comment (HonkingGoose) [reply docs/development/design-decisions.md:10]:
```suggestion
Right now we do not have a stable API.
```

File: docs/development/design-decisions.md [hunk 4]
```diff
@@ -2,6 +2,17 @@
 
 This file documents the design choices as well as configuration options.
 
+## Intended use by end-users
+
+Renovate is intended to be used as an app.
```
Comment (rarkins) [inline docs/development/design-decisions.md:7]:
```suggestion
The Renovate repository/package is intended to be used as a CLI-based application.
```

File: docs/development/design-decisions.md [hunk 5]
```diff
@@ -2,6 +2,17 @@
...
+It should not be used as a library, because it lacks a stable API.
```
Comment (rarkins) [inline docs/development/design-decisions.md:8]:
```suggestion
It should not be used downstream as a library, because it lacks a stable API.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/development/design-decisions.md [hunk 6]
```diff
@@ -2,6 +2,17 @@
 
 This file documents the design choices as well as configuration options.
 
+## Intended use by end-users
+
+The Renovate repository/package is intended to be used as a CLI-based application.
+It should not be used downstream as a library, because it lacks a stable API.
+
+Right now we do not have a stable API.
+The API and documentation would need a major overhaul each time we refactor our code.
```
Comment (HonkingGoose) [inline docs/development/design-decisions.md:11]:
```suggestion
```

We can probably get rid of these 3 lines now?

**Review by rarkins — APPROVED** [review]

**Pull request merged**